src/Umbraco.Infrastructure/PropertyEditors/DateTimePropertyEditor.cs

```diff
@@ -23,15 +25,22 @@ namespace Umbraco.Cms.Core.PropertyEditors
     public class DateTimePropertyEditor : DataEditor
     {
         private readonly IIOHelper _ioHelper;
+        private readonly IEditorConfigurationParser _editorConfigurationParser;
 
+        public DateTimePropertyEditor(IDataValueEditorFactory dataValueEditorFactory, IIOHelper ioHelper)
+            : this(dataValueEditorFactory, ioHelper, StaticServiceProvider.Instance.GetRequiredService<IEditorConfigurationParser>())
+        {
+            _ioHelper = ioHelper;
+        }
```

Comment:
We probably want to obsolete this as well, and since we're calling the other constructor we don't have to set `_ioHelper` in this constructor as well.